src/main/java/com/endava/internship/warmup/domain/service/ArrayProcessorWithForLoops.java

```diff
@@ -126,7 +260,14 @@ public int findSecondMax(final int[] input) {
      */
     @Override
     public void validateForMatrixMultiplication(int[][] leftMatrix, int[][] rightMatrix) throws NullPointerException, IllegalArgumentException {
-        throw new UnsupportedOperationException("Method not implemented yet");
+        if (matrixDimensionsAreIllegal(leftMatrix)
+                || matrixDimensionsAreIllegal(rightMatrix)
+                || leftMatrix[0].length != rightMatrix.length) {
+            throw new IllegalArgumentException("Matrices are of illegal dimensions!");
+        }
+        if (isNull(leftMatrix) || isNull(rightMatrix)) {
```

Comment:
The matrix will not be checked here, because their elements(length) are already checked in previous if methods, and if they are null, it will throw an NPE before you check them here

Reply:
Swapped checks